src/messagetags.js

```diff
@@ -33,11 +33,30 @@ function encodeValue(value) {
     return value.replace(val_lookup, m => vals_map[m] || '');
 }
 
+function splitN(inputString, separator, n) {
+    const ret = [];
+    const maxParts = n - 1;
+
+    const allParts = inputString.split(separator);
+
+    for (const [i, v] of allParts.entries()) {
+        if (i < maxParts) {
+            ret.push(v);
+        } else {
+            const rest = allParts.slice(i).join(separator);
+            ret.push(rest);
+            break;
+        }
+    }
+
+    return ret;
+}
+
 function decode(tag_str) {
     let tags = Object.create(null);
 
     tag_str.split(';').forEach(tag => {
-        let parts = tag.split('=');
+        let parts = splitN(tag, '=', 2);
```

Comment:
What about `indexOf('=')` and using `tag.splice`?

Reply:
You're right, it's wasteful to let String.prototype.split do unnecessary work.